integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -395,17 +396,22 @@ private int openInternal(String path, FuseFileInfo fi) {
         return -ErrorCodes.EOPNOTSUPP();
       }
     }
-
+    long fd = mNextOpenFileId.getAndIncrement();
     fi.fh.set(fd);
     try {
-      if (openAction == OpenAction.WRITE_ONLY) {
-        if (status != null) {
-          OpenFlags openFlags = OpenFlags.valueOf(flags);
-          if (openFlags == OpenFlags.O_CREAT || openFlags == OpenFlags.O_EXCL) {
-            return -ErrorCodes.EEXIST();
-          }
-          mFileSystem.delete(uri);
+      // if this is an write operation, and file already exist
+      // we according to O_TRUNC flag to decide whether remove old file
+      // if not specify O_TRUNC, because alluxio can't overwrite part of content of a file, return error for user
+      // attention: according to posix standard, we can't remove file only specify O_WRONLY or O_RDWR flag
+      if (openAction != OpenAction.READ_ONLY && status != null) {
+        if((flags & OpenFlags.O_TRUNC.longValue()) == 0) {
+          LOG.error("Cannot write exist file {} without O_TRUNC flag specify, flags:{}", path, flags);
```

Comment:
For the workloads provided by @liuyongqing, 
when Alluxio file exists, the file is open with 0x8001.
The workload is `getattr() - file exist - open(0x8001) - write (offset = 0, len = 19) - release()`
```
  // w Open file for writing. The file is created (if it does not exist) or
  // truncated (if it exists).
  // (Decimal): 32769 (Octal): 00000100001 (Hexadecimal): 00008001
  // (Binary ): 00000000000000001000000000000001
  private static final int WRITE = 0x8001;
```
0x8001 includes the meaning of truncating.
Alluxio doesn't obey posix standard that it cannot supports file overwriting or appending, that's why we do a delete and create combination.
@liuyongqing what's your worries about this workload, what's the expected behavior of this workload? Alluxio error out directly?